spec/std/yaml/serializable_spec.cr

```diff
@@ -914,6 +937,48 @@ describe "YAML::Serializable" do
     end
   end
 
+  describe "serializes YMAL with presence markers and ignore_serialize" do
+    context "ignore_serialize is set to a method which returns true when value is nil or empty string" do
+      it "ignores field when value is empty string" do
+        yaml = YMALAttrWithPresenceAndIgnoreSerialize.from_yaml(%({"first_name": ""}))
+        yaml.first_name_present?.should be_true
+        yaml.to_yaml.should eq("--- {}\n")
+      end
+
+      it "ignores field when value is nil" do
+        yaml = YMALAttrWithPresenceAndIgnoreSerialize.from_yaml(%({"first_name": null}))
+        yaml.first_name_present?.should be_true
+        yaml.to_yaml.should eq("--- {}\n")
+      end
+    end
+
+    context "ignore_serialize is set to conditional expressions 'last_name.nil? && !last_name_present?'" do
+      it "emits null when value is null and @last_name_present is true" do
+        yaml = YMALAttrWithPresenceAndIgnoreSerialize.from_yaml(%({"last_name": null}))
+        yaml.last_name_present?.should be_true
+        yaml.to_yaml.should eq("---\nlast_name: \n")
```

Comment:
Perhaps it would be better to strip the trailing whitespace in order to achieve cross-platform quirks?

```suggestion
        yaml.to_yaml.rstrip.should eq("---\nlast_name:")
```